packages/product-exporter/test/main.spec.js

```diff
@@ -0,0 +1,167 @@
+      expect(productExporter.logger).toBeDefined()
+      expect(productExporter.client).toBeDefined()
+      expect(productExporter.exportConfig).toBeDefined()
+    })
+  })
+
+  describe('::run', () => {
+    it('prepare the output stream and pass to `_getProducts`', async() => {
+      productExporter._getProducts = jest.fn(() => Promise.resolve())
+      const outputStream = streamtest['v2'].toText(() => {})
+      await productExporter.run(outputStream)
+      expect(productExporter._getProducts).toBeCalled()
+      expect(productExporter._getProducts).not.toBeCalledWith(outputStream)
+    })
+
+    it('should emit `error` on output stream if error occurs', (done) => {
+      productExporter._getProducts = jest.fn(() =>
+        Promise.reject(new Error('error occured')),
+      )
+
+      const outputStream = streamtest['v2'].toText((error, result) => {
+        expect(error.message).toBe('error occured')
+        expect(result).toBeUndefined()
+        done()
+      })
+      productExporter.run(outputStream)
+    })
+  })
+
+  describe('::_getProducts', () => {
+    let processMock
+    const outputStream = {
+      emit: () => {},
+      end: jest.fn(),
+    }
+    beforeEach(() => {
+      const sampleResult = {
+        body: {
+          results: [],
+        },
+      }
+      processMock = jest.fn((request, processFn) => (
+        processFn(sampleResult).then(() => Promise.resolve())
+      ))
+      productExporter.client.process = processMock
+    })
+
+    it('should fetch products using `process` method', () => {
+      productExporter._getProducts(outputStream)
+      expect(processMock).toHaveBeenCalledTimes(1)
+      expect(processMock.mock.calls[0][0])
+        .toEqual({
+          uri: oneLineTrim`
+            /project-key/product-projections?staged=true
+            &expand=something&where=foo%3Dbar&limit=5`,
+          method: 'GET',
+          headers: {
+            Authorization: 'Bearer myAccessToken',
+          },
+        })
+    })
+
+    it('should pass the products and the stream to the writer method', () => {
+      const spy = jest.spyOn(ProductExporter, '_writeEachProduct')
+
+      productExporter._getProducts(outputStream)
+      expect(spy).toBeCalled()
+      spy.mockRestore()
+    })
+
+    it('should close stream after writing data', async() => {
+      productExporter.client.process = jest.fn(() => Promise.resolve())
+      await productExporter._getProducts(outputStream)
+      expect(outputStream.end).toBeCalled()
+    })
+  })
+
+  describe('::_buildProductProjectionsUri', () => {
+    const projectKey = 'my-project-key'
+    const exportConfig = {
+      staged: true,
+      batch: 5,
+      predicate: 'foo=bar',
+      expand: ['someReference', 'anotherReference'],
+    }
+
+    it('should build uri with query options', () => {
+      const expectedUri = oneLineTrim`
+        /my-project-key/product-projections?staged=true&expand=someReference
```

Comment:
nitpick if you are using oneLinetrim, you might as well separate the slashes into an new line:

```js
oneLineTrim`
/all
/the
/values
/trimmed
/into
/one
/line
`
```

Reply:
We still need to make it readable though not one word per line.